### DISCUSSION TITLE
Visualize acoustic events, including system noise, relative to tides +

### DISCUSSION THREAD

**Discussion by scottveirs** [orcasound/bioacoustic-dashboard#8] [Ideas]
On the weekly Orcasound standup call, Val @veirs , Dave Bain @dbainj1 , and I discussed how to monitor and log system performance across Orcasound locations, especially during this winter period when high tides and storms conspire to destroy our hydrophones and electronics. This led to question we'd like to discuss: as Orcasound hydrophone hosts and network coordiantors, how could we visualize the emerging sources of data related to hydrophone hardware performance, including:

1. Human detections of system noise via [live.orcasound.net](live.orcasound.net) generated when a user of the live-listening web app (v3) selects the `Report Sound` button, then the `Other` category, and describes system noise in the comment field. These can be reviewed and replayed by searching through the new [Orcasound v3 reports list](https://live.orcasound.net/reports) for Other and/or "system noise."
2. OrcaHello detections of system noise -- false positives from the binary SRKW call classifier that have been tagged with the `system` label. These can be reviewed by visiting the OrcaHello moderator portal dashboard, selecting a time period of interest, and than looking through the candidates that have the `system` tag. 

Both of these data sources are increasingly accessible programmatically through their respective APIs...

Here are two ideas for how such system noise indicators could be utilized and organized:
- Translate them into a csv file for visualization in the SeaStats `Acoustic events` view. 
<img width="1744" alt="Screenshot 2023-12-06 at 11 58 46 AM" src="https://github.com/orcasound/bioacoustic-dashboard/assets/14044595/57e9bc5d-a6f4-4444-8966-ed51c0f73f93">
There could be two categories of acoustic signal types (aka `system noise` in this case, not `Call type`): human-detected and machine-detected. Following Ben's approach to grouping bioacoustic events from whales (`calls`), this may benefit from the use of Density Based Clustering for Applications with Noise ([DBCAN](https://en.wikipedia.org/wiki/DBSCAN) | [Ester et al., 1996](https://cdn.aaai.org/KDD/1996/KDD96-037.pdf))
- Use their associated audio data to train a system noise classifier, then run the model through the raw audio data near the label(s) until the bounds of the system noise are found (and perhaps the nature/cause of the noise is determined!)

From the trouble-shooting perspective of a hydrophone host, an even better visualization would juxtapose the indicators of system noise with **_environmental factors that often govern hydrophone system performance: tidal height, tidal current (velocity, but also direction), wind speed (mean and peak/gust), and wave/wake statistics._** For example, at Sunset Point the noise level increased with tidal height; there the ladder to which the hydrophone is attached would gurgle whenever the tide reached and then filled one of the hollow rungs! At other Orcasound location, we suspect system noise increases when a falling tide disconnects or weakens a saltwater ground. At other locations, we might expect both system noise and geophony to increase during periods of high current velocity (cable thrumming, intertidal connection vibrating, flow noise at the hydrophone). Similar noise increases are commonly caused during a storm when wind-driven waves break or move the hydrophone/cable or when rain falls on the sea surface.

**Are there open access sources for such environmental time series near the Orcasound nodes?**

Let's discuss, but here is a short starter-list:

- **Tidal height**: work with Jeff Dairiki, creator of our favorite [tidal height and current plots](https://www.dairiki.org/tides/daily.php)
- **Current vectors**: figure out how to access the daily forecasts of (surface) currents from the [LiveOcean model](https://github.com/parkermac/LO)
- **Wind speed**: the nearest weather station and/or NOAA buoy?
- **Wave statistics**: the nearest NOAA buoy? Radar?

**Assuming you had the system noise and environmental data in hand, how would to analyze/cluster the acoustic events and then display the environmental variables (e.g. as layers like the sun, moon, and data gap ones featured in the current SeaStats view)?!**

**Comment by dbainj1** [comment]
Another problem we have, on Point Robinson at least, is Internet failure.  We were just down for a couple days while the mobile hotspot was down.  We've also had down time while my home Internet has been down, blocking the YouTube to Orcasound bridge.We also have the brief down time when the Pis are rebooting.At Sunset Bay, we have down time while the tide is too low.Power outages can be another problem.No signal would be an appropriate tag for human listeners.  It could be helpful to keep a written log that notes when down time is first detected and when the problem is resolved.  Then someone or somethingcould search for the actual beginning and end of the problem.We'd also want a system down symbol on the event plot, so viewers will no that absence of data is not necessarily the same thing as absence of events.  Things like gurgling are good to note as we try to optimize system performance, but they don't necessarily prevent detections, so could go on a separate performance plot instead of the event plot.DaveSent from Samsung Galaxy smartphone.
-------- Original message --------From: Scott Veirs ***@***.***> Date: 12/6/23  12:24 PM  (GMT-08:00) To: orcasound/bioacoustic-dashboard ***@***.***> Cc: dbainj1 ***@***.***>, Mention ***@***.***> Subject: [orcasound/bioacoustic-dashboard] Visualize acoustic events,
  including system noise, relative tides and wind (Discussion #8) 
On the weekly Orcasound standup call, Val @veirs , Dave Bain @dbainj1 , and I discussed how to monitor and log system performance across Orcasound locations, especially during this winter period when high tides and storms conspire to destroy our hydrophones and electronics. This led to question we'd like to discuss: as Orcasound hydrophone hosts and network coordiantors, how could we visualize the emerging sources of data related to hydrophone hardware performance, including:

Human detections of system noise via live.orcasound.net generated when a user of the live-listening web app (v3) selects the Report Sound button, then the Other category, and describes system noise in the comment field. These can be reviewed and replayed by searching through the new Orcasound v3 reports list for Other and/or "system noise."
OrcaHello detections of system noise -- false positives from the binary SRKW call classifier that have been tagged with the system label. These can be reviewed by visiting the OrcaHello moderator portal dashboard, selecting a time period of interest, and than looking through the candidates that have the system tag.

Both of these data sources are increasingly accessible programmatically through their respective APIs...
Here are two ideas for how such system noise indicators could be utilized and organized:

Translate them into a csv file for visualization in the SeaStats Acoustic events view.

Screenshot.2023-12-06.at.11.58.46.AM.png (view on web)
There could be two categories of acoustic signal types (aka `system noise` in this case, not `Call type`): human-detected and machine-detected. Following Ben's approach to grouping bioacoustic events from whales (`calls`), this may benefit from the use of Density Based Clustering for Applications with Noise ([DBCAN](https://en.wikipedia.org/wiki/DBSCAN) | [Ester et al., 1996](https://cdn.aaai.org/KDD/1996/KDD96-037.pdf))
- Use their associated audio data to train a system noise classifier, then run the model through the raw audio data near the label(s) until the bounds of the system noise are found (and perhaps the nature/cause of the noise is determined!)
From the trouble-shooting perspective of a hydrophone host, an even better visualization would juxtapose the indicators of system noise with environmental factors that often govern hydrophone system performance: tidal height, tidal current (velocity, but also direction), wind speed (mean and peak/gust), and wave/wake statistics. For example, at Sunset Point the noise level increased with tidal height; there the ladder to which the hydrophone is attached would gurgle whenever the tide reached and then filled one of the hollow rungs! At other Orcasound location, we suspect system noise increases when a falling tide disconnects or weakens a saltwater ground. At other locations, we might expect both system noise and geophony to increase during periods of high current velocity (cable thrumming, intertidal connection vibrating, flow noise at the hydrophone). Similar noise increases are commonly caused during a storm when wind-driven waves break or move the hydrophone/cable or when rain falls on the sea surface.
Are there open access sources for such environmental time series near the Orcasound nodes?
Let's discuss, but here is a short starter-list:

Tidal height: work with Jeff Dairiki, creator of our favorite tidal height and current plots
Current vectors: figure out how to access the daily forecasts of (surface) currents from the LiveOcean model


Assuming you had the system noise and environmental data in hand, how would to analyze/cluster the acoustic events and then display the environmental variables (e.g. as layers like the sun, moon, and data gap ones featured in the current SeaStats view)?!
